### PR TITLE
Changes in nfp fwaas plugin with monkey patched router methods, creat…

### PR DESCRIPTION
**Pull request by ashutosh-mishra**
…e_firewall

### CODE REVIEW

**Comment by ashutosh-mishra** [comment]
@njagadish @hemanthravi @songole

File: gbpservice/nfp/service_plugins/firewall/nfp_fwaas_plugin.py [hunk 1]
```diff
@@ -33,28 +39,53 @@ def _get_routers_for_create_firewall(self, tenant_id, context, firewall):
...
-                return []
-            else:
-                # some router(s) provided.
-                # Dont validate this condition
-                # self.validate_firewall_routers_not_in_use(context,
-                # router_ids)
-                return router_ids
+            return tenant_id
+
+    def nfp_create_firewall(self, context, firewall, status=None):
```
Comment (songole) [inline gbpservice/nfp/service_plugins/firewall/nfp_fwaas_plugin.py]: You are overwriting a method in super class. Name should match.
Remove nfp_ prefix. Just call create_firewall(...)

File: gbpservice/nfp/service_plugins/firewall/nfp_fwaas_plugin.py [hunk 2]
```diff
@@ -33,28 +39,53 @@ def _get_routers_for_create_firewall(self, tenant_id, context, firewall):
...
+                    tenant_id=tenant_id,
+                    name=fw['name'],
+                    description=fw['description'],
+                    firewall_policy_id=fw['firewall_policy_id'],
+                    admin_state_up=fw['admin_state_up'],
+                    status=status)
+            context.session.add(firewall_db)
+        return self._make_firewall_dict(firewall_db)
+
+    n_firewall.Firewall_db_mixin.create_firewall = nfp_create_firewall
```
Comment (songole) [inline gbpservice/nfp/service_plugins/firewall/nfp_fwaas_plugin.py]: This patch is not needed with the above name overwrite.

File: gbpservice/nfp/service_plugins/firewall/nfp_fwaas_plugin.py [hunk 3]
```diff
@@ -33,28 +39,53 @@ def _get_routers_for_create_firewall(self, tenant_id, context, firewall):
...
+        """
+        pass
+
+    def update_firewall_routers(self, context, fw):
+        """Update the firewall with new routers.
+
+        This involves removing existing router associations and replacing
+        it with the new router associations provided in the update method.
+        """
+        return fw
```
Comment (songole) [inline gbpservice/nfp/service_plugins/firewall/nfp_fwaas_plugin.py:98]: Now N-S L3 patch can be added here as monkey patch. Looks like code is different between liberty and mitaka.

For mitaka
from neutron.db import l3_db
def nfp_get_router_floatingip()
"""existing code"""
"""lines 135-141 from N-S extraroute patch"
"""existing Notfound exception"
}
""" Lines 1-127 from extraroute patch"

l3_db.L3_NAT_dbonly_mixin.get_router_for_floatingip = nfp_get_router_for_floatingip

**Comment by ashutosh-mishra** [comment]
Closing this, as raised another PR - https://github.com/oneconvergence/group-based-policy/pull/444

**Pull request closed without merging**